Run3Detector/analysis/utilities/milliqanProcessor.py

```diff
@@ -10,14 +12,96 @@
 
 class milliqanProcessor():
 
-    def __init__(self, filelist, branches, schedule=None, cuts=None, plotter=None, max_events=None):
+    def __init__(self, filelist, branches, schedule=None, cuts=None, plotter=None, max_events=None, runQualityOverride=False, qualityLevel="tight"):
+        self.qualityLevelString = qualityLevel
+        self.runQualityOverride = runQualityOverride
+        
+        #Converting the quality level to an integer
+        self.qualityDict = {"single_trigger": -1, "loose": 0, "medium": 1, "tight": 2}
+        if self.qualityLevelString not in self.qualityDict.keys():
+            raise Exception("Quality level '{0}' not recognized".format(self.qualityLevelString))
+        print("Chosen quality level: ", self.qualityLevelString)
+        self.qualityLevel = self.qualityDict[self.qualityLevelString]
+        
+        #Checks the filelist against goodRuns.json
         self.filelist = filelist
+        self.fileChecker() 
+        
         self.branches = branches
         self.mqSchedule = schedule
         #self.mqCuts = cuts
         #self.plotter = plotter
         self.max_events = max_events
 
+    #Get rid of the strings and make a dictionary so that it's easier to debug
+    def fileChecker(self):
+        #goodJson_array = ak.from_json(pathlib.Path("../goodRunTools/goodRunsMerged.json"))
+        goodJson_array = ak.from_json(pathlib.Path("../../configuration/barConfigs/goodRunsList.json"))
+        data = ak.Array(goodJson_array['data'])
+        goodJson = ak.zip({
+            'run': data[:, 0],
+            'file': data[:, 1],
+            'loose': data[:, 2],
+            'medium': data[:, 3],
+            'tight': data[:, 4],
+            'single_trigger': data[:, 5]
+        }, depth_limit=1)
+        
+        for filepath in self.filelist:
+            filename = os.path.basename(filepath)
+            parts = filename.split('_')
+            run_number, file_number = parts[1].replace("Run","").split('.')
+            matching_goodJson = goodJson[(goodJson['run'] == int(run_number)) & (goodJson['file'] == int(file_number))]
+            
+            #Establishes the quality level of the run
+            runQualityLevel = -1
+            if len(matching_goodJson) == 0:
+                pass
+            elif (matching_goodJson['tight'] == True) and (matching_goodJson['medium'] == True) and (matching_goodJson['loose'] == True):
```

Comment:
to avoid all of this checking can we just have one line that removes the file if it does not pass

`if (matching_goodJson[self.qualityLevel] != True:
    self.filelist.remove(filename)`

Reply:
I think all of the below can be removed